server/controllers/authentication.coffee

```diff
@@ -93,14 +93,19 @@ module.exports.saveUnauthenticatedUser = (req, res, next) ->
         User.all (err, users) ->
             return next new Error err if err
             # if existing user document with password -> request rejected
-            if users[0]?.password
+            if users[0]?.password and not requestData.password?
                 error        = new Error 'Not authorized'
                 error.status = 401
                 return next error
             else if users.length
                 users[0].merge userToSave, (err) ->
                     return next new Error err if err
-                    res.status(200).send(result: 'User data correctly updated')
+                    if requestData.password?
+                        return Authentication.authenticate(req, res, next)
```

Comment:
You've made a good point. As discussed with @poupotte, we will first validate a version of the onboarding where the user is transparently authenticated.
But I agree with you : authenticate at the end of a function supposed to only update the user property is an anti-pattern. Where do you think we may do the authentication ? In a separate method added in the route configuration ? Something like:
```coffeescript
'register/password':
        put: [utils.isAuthenticated, utils.isNotRegistered, auth.saveAuthenticatedUser, auth.authenticate]
```